server/service/externalsvc/okta.go

```diff
@@ -0,0 +1,79 @@
+package externalsvc
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/fleetdm/fleet/v4/pkg/fleethttp"
+)
+
+var ErrInvalidGrant = errors.New("the credentials provided were invalid")
+
+type Okta struct {
+	BaseURL      string
+	ClientID     string
+	ClientSecret string
+}
+
+// oktaError is the response body for requests with errors coming from Okta.
+type oktaError struct {
+	Error       string `json:"error"`
+	Description string `json:"error_description"`
+}
+
+// ROPLogin performs a login using the "Resource Owner Password Flow" as
+// specified by RFC 6749 and described in
+// https://developer.okta.com/docs/guides/implement-grant-type/ropassword/main/
+func (o *Okta) ROPLogin(ctx context.Context, username, password string) error {
+	params := url.Values{
+		"username":   []string{username},
+		"password":   []string{password},
+		"scope":      []string{"openid"},
+		"grant_type": []string{"password"},
+	}
+	req, err := http.NewRequestWithContext(
+		ctx, "POST", o.BaseURL,
+		strings.NewReader(params.Encode()),
+	)
+	if err != nil {
+		return err
```

Comment:
Or `fmt.Errorf`'d, I notice that we don't do that in the other external services, not sure why...

Reply:
I wasn't sure either so I mimicked what was there, but you're right: that's not a good reason!

I used `fmt.Errorf` because there's an import cycle with `ctxerr.Wrap` (maybe that's why errors in this package are not wrapped?)